### PR TITLE
Fix issues with mixpanel events

### PR DESCRIPTION
**Pull request by zerts**

### CODE REVIEW

File: src/shared/analytics/mixpanel.ts [hunk 1]
```diff
@@ -33,6 +34,19 @@ class DeviceIdStore extends PersistentStore<string | undefined> {
   }
 }
 
+function omitNullParams<T extends Record<string, unknown>>(
+  params: T
+): Partial<T> {
+  return produce(params, (draft) => {
+    const keys = Object.keys(draft);
+    for (const key of keys) {
+      if (draft[key] == null) {
```
Comment (everdimension) [inline src/shared/analytics/mixpanel.ts:43]: what about empty arrays?

Comment (zerts) [reply src/shared/analytics/mixpanel.ts:43]: We discussed only `name: null` cases as bad

Comment (zerts) [reply src/shared/analytics/mixpanel.ts:43]: This is because if Mixpanel's limitations

**Review by vyorkin — APPROVED** [review]

File: src/shared/analytics/mixpanel.ts [hunk 2]
```diff
@@ -33,6 +34,19 @@ class DeviceIdStore extends PersistentStore<string | undefined> {
   }
 }
 
+function omitNullParams<T extends Record<string, unknown>>(
+  params: T
+): Partial<T> {
+  return produce(params, (draft) => {
+    const keys = Object.keys(draft);
+    for (const key of keys) {
```
Comment (everdimension) [inline src/shared/analytics/mixpanel.ts:42]: White not `for (const key in draft)`?

Comment (zerts) [reply src/shared/analytics/mixpanel.ts:42]: No reason, just forgot this syntax

File: src/shared/analytics/mixpanel.ts [hunk 3]
```diff
@@ -155,9 +154,9 @@ class MixpanelApi {
       url.searchParams.append('verbose', '1');
     }
     url.searchParams.append('ip', '1');
-    const payload = omitNullParams({
+    const payload = {
       event,
-      properties: {
+      properties: omitNullParams({
```
Comment (everdimension) [inline src/shared/analytics/mixpanel.ts:159]: 👍

**Review by everdimension — APPROVED** [review]

**Pull request merged**